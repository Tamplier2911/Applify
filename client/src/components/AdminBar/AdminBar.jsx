// import "./AdminBar.scss";
import React from "react";

// JS Rendering CSS
import {
  ProfilePortraitAdminBar,
  ProfilePortraitAdminUL,
  ProfilePortraitAdminLI,
  ProfilePortraitAdminLink,
  ProfilePortraitAdminUsersSVG,
  ProfilePortraitAdminMessagesSVG,
  ProfilePortraitAdminPostsSVG,
  ProfilePortraitAdminFeedsSVG
} from "./AdminBarStyles.js";

const AdminBar = () => {
  return (
    <ProfilePortraitAdminBar>
      <ProfilePortraitAdminUL>
        <ProfilePortraitAdminLI>
          <ProfilePortraitAdminLink to="profile/users">
            <ProfilePortraitAdminUsersSVG />
          </ProfilePortraitAdminLink>
        </ProfilePortraitAdminLI>
        <ProfilePortraitAdminLI>
          <ProfilePortraitAdminLink to="profile/messages">
            <ProfilePortraitAdminMessagesSVG />
          </ProfilePortraitAdminLink>
        </ProfilePortraitAdminLI>
        <ProfilePortraitAdminLI>
          <ProfilePortraitAdminLink to="profile/blogs">
            <ProfilePortraitAdminPostsSVG />
          </ProfilePortraitAdminLink>
        </ProfilePortraitAdminLI>
        <ProfilePortraitAdminLI>
          <ProfilePortraitAdminLink to="profile/feeds">
            <ProfilePortraitAdminFeedsSVG />
          </ProfilePortraitAdminLink>
        </ProfilePortraitAdminLI>
      </ProfilePortraitAdminUL>
    </ProfilePortraitAdminBar>
  );
};

export default AdminBar;
