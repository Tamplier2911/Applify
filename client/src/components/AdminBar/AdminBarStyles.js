import styled, { css } from "styled-components";
import { Link } from "react-router-dom";

// svg
import { ReactComponent as UsersSVG } from "../../assets/svg/adminpage-users.svg";
import { ReactComponent as MessagesSVG } from "../../assets/svg/adminpage-messages.svg";
import { ReactComponent as BlogsSVG } from "../../assets/svg/adminpage-blogs.svg";
import { ReactComponent as FeedsSVG } from "../../assets/svg/adminpage-feeds.svg";

const SVGSharedStyles = css`
  width: 3rem;
  height: 3rem;
  fill: var(--cl-font);
  transition: 0.3s fill;

  @media only screen and (max-width: 200px) {
    width: 2rem;
    height: 2rem;
  }

  &:hover {
    fill: var(--cl-primary);
  }
`;

export const ProfilePortraitAdminBar = styled.div`
  display: grid;
`;

export const ProfilePortraitAdminUL = styled.ul`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
`;

export const ProfilePortraitAdminLI = styled.li`
  display: grid;
  align-items: center;
  justify-items: center;
  list-style: none;
`;

export const ProfilePortraitAdminLink = styled(Link)``;

export const ProfilePortraitAdminUsersSVG = styled(UsersSVG)`
  ${SVGSharedStyles}
`;

export const ProfilePortraitAdminMessagesSVG = styled(MessagesSVG)`
  ${SVGSharedStyles}
`;

export const ProfilePortraitAdminPostsSVG = styled(BlogsSVG)`
  ${SVGSharedStyles}
`;

export const ProfilePortraitAdminFeedsSVG = styled(FeedsSVG)`
  ${SVGSharedStyles}
`;
