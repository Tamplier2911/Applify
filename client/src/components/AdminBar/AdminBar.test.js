import React from "react";

// component
import AdminBar from "./AdminBar";

// utils
import { setShallow, findByTestAttr } from "../../tests/testUtils";

describe("<AdminBar />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setShallow(<AdminBar />);
  });

  it("renders without error", () => {
    const adminBar = findByTestAttr(wrapper, "admin-bar");
    expect(adminBar.length).toBe(1);
  });

  it("renders children link - users", () => {
    const adminBarUsersLink = findByTestAttr(wrapper, "admin-bar-users");
    expect(adminBarUsersLink.length).toBe(1);
  });

  it("renders children link - messages", () => {
    const adminBarMessagesLink = findByTestAttr(wrapper, "admin-bar-messages");
    expect(adminBarMessagesLink.length).toBe(1);
  });

  it("renders children link - blogs", () => {
    const adminBarMessagesLink = findByTestAttr(wrapper, "admin-bar-blogs");
    expect(adminBarMessagesLink.length).toBe(1);
  });

  it("renders children link - feeds", () => {
    const adminBarBlogsLink = findByTestAttr(wrapper, "admin-bar-feeds");
    expect(adminBarBlogsLink.length).toBe(1);
  });

  it("matches snapshot", () => {
    const adminBar = findByTestAttr(wrapper, "admin-bar");
    expect(adminBar).toMatchSnapshot();
  });
});
