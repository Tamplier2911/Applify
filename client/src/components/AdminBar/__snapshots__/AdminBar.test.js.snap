// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminBar /> matches snapshot 1`] = `
<styled.div
  data-test="admin-bar"
>
  <styled.ul>
    <styled.li>
      <Styled(Link)
        data-test="admin-bar-users"
        to="profile/users"
      >
        <Styled(Component) />
      </Styled(Link)>
    </styled.li>
    <styled.li>
      <Styled(Link)
        data-test="admin-bar-messages"
        to="profile/messages"
      >
        <Styled(Component) />
      </Styled(Link)>
    </styled.li>
    <styled.li>
      <Styled(Link)
        data-test="admin-bar-blogs"
        to="profile/blogs"
      >
        <Styled(Component) />
      </Styled(Link)>
    </styled.li>
    <styled.li>
      <Styled(Link)
        data-test="admin-bar-feeds"
        to="profile/feeds"
      >
        <Styled(Component) />
      </Styled(Link)>
    </styled.li>
  </styled.ul>
</styled.div>
`;
